test-dts/defineComponent.test-d.tsx

```diff
@@ -258,6 +263,14 @@ describe('with object props', () => {
 
   expectType<Component>(MyComponent)
 
+  expectType<typeof props>(MyComponent.props)
```

Comment:
I also think we need to have an expect-error here, probably something like:
```ts
  // @ts-expect-error it should be an object and not any
  expectType<[]>(MyComponent.props)
```

Reply:
done